src/microsim/schema/emission.py

```diff
@@ -56,84 +37,85 @@ def ec_to_cross_section(ec: Any) -> pint.Quantity:
+        raise NotImplementedError("Fluorophore has no excitation spectrum.")
 
-    # find the subset of the spectra with overlapping wavelengths
-    ex_spectrum, filter_spectrum = get_overlapping_spectra(
-        fluor_ex_spectrum, filter_spectrum
-    )
+    if (ext_coeff := fluor.extinction_coefficient) is None:
+        raise NotImplementedError("Fluorophore has no extinction coefficient.")
 
-    # wavelengths in nm
-    wavelengths = ex_spectrum[:, 0] * ureg.nm
-    # power in units of W x cm-2 (irradiance), we're making up constant 100 for now...
-    irradiance = filter_spectrum[:, 1] * _ensure_quantity(light_power, "W/cm^2")
-    # scale ex_spectrum by extinction coefficient
-    # note, fluor.extinction_coefficient is already a pint Quantity of units 1/M/cm
-    ext_coeff = ex_spectrum[:, 1] * fluor.extinction_coefficient
-
-    # calculate the number of photons hitting a fluorophore per second
-    exc_rate = fluorophore_photon_flux(wavelengths, irradiance, ext_coeff)
+    # TODO: derive light power from model
+    irradiance = ex_filter_spectrum * _ensure_quantity(light_power, "W/cm^2")
+    cross_section = fluor_ex_spectrum * ec_to_cross_section(ext_coeff)
+    power_absorbed = cross_section * irradiance
+    excitation_rate = power_absorbed / energy_per_photon(power_absorbed.wavelength)
+    return excitation_rate
 
+    # TODO
     # if the fluorophore has a lifetime, calculate the effective excitation rate
     # based on the fraction of time in the excited state
     # (i.e. at some point, we'll hit ground state depletion and saturation effects)
     if (lifetime := getattr(fluor, "lifetime", None)) is not None:
         lifetime = _ensure_quantity(lifetime, "ns")
         # Calculate the fraction of time in the excited state
-        f_excited = exc_rate * lifetime
+        f_excited = excitation_rate * lifetime
         # Calculate the fraction of time in the ground state
         f_ground = 1 - f_excited
         # Calculate the effective excitation rate
-        exc_rate = exc_rate * f_ground
+        excitation_rate = excitation_rate * f_ground
+
+    return excitation_rate
+
+
+def get_emission_events(
+    channel: OpticalConfig,
+    fluor: Fluorophore,
+    *,
+    light_power: float = 100,
+) -> Spectrum:
+    # get the emission events for the given fluorophore
+    if (channel_ex := channel.excitation) is None:
+        raise NotImplementedError("Channel without excitation spectrum?")
+
+    if not (fluor_em_spectrum := fluor.emission_spectrum):
+        raise NotImplementedError("Fluorophore has no emission spectrum.")
+
+    ex_rate = get_excitation_rate(channel_ex.spectrum, fluor, light_power=light_power)
+
+    # convert the excitation rate to an emission rate, by matching the integral
+    # of the excitation and emission spectra
+    # I *think* this is correct, but I'm not 100% sure
+    scaling_factor = ex_rate.integral() / fluor_em_spectrum.integral()
```

Comment:
Under the assumption that the emission spectra overall shape does not change, I think the scaling_factor should be computed as `scaling_factor = ex_rate.integral()/ fluor.excitation_spectrum.integral()`.  
It is so because, if there were no excitation filters and light containing all relevant wavelengths was incident, the photons which will excite the flurophores should form the pattern of excitation_spectrum of the fluorophore and we will get emission matching fluorophore's emission spectra. Now, due to excitation filters and the incident light's wavelength range, the excitation spectra is curtailed by `scaling_factor` and so the emission spectra should be accordingly curtailed. Ofcourse, this is just my hypothesis. Would love to know if I'm wrong.

Reply:
But it now seems to me that ex_rate.integral() takes into account the laser power, incidence duration and so your motivation is to transfer that factor into the emission. So, with your formulation, you are assuming that emission spectra does not change and therefore you simply scale it appropriately. I think that also makes sense to me now.

Reply:
Yeah that was my thinking. Exc_rate is the wavelength dependent frequency of absorption events, accounting already for the fluorophore excitation spectrum, extinction coefficients, and illumination power.  So, putting aside quantum yield, which comes in a moment, I think we just want to convert the sum total of excitation events (per second) to an equal number of emission events (per second).  This makes the assumption that emission wavelength is independent of the wavelength of the exciting photon (which is a fine assumption for our purposes).